vm/src/macros.rs

```diff
@@ -35,6 +35,34 @@ macro_rules! type_check {
     };
 }
 
+#[macro_export]
+macro_rules! check_but_allow_none {
```

Comment:
This is a rather specific macro. The only usecase is now to allow globals to be none. Perhaps add this logic into the make_scope function and remove the macro. Do you foresee more usage of this macro outside checking the global for none?

Reply:
I'm imagining we'll find other built-ins that use this argument can be None argument, though I haven't seen them yet. I think we need to do the check before we compile the code (should raise TypeError before SyntaxError), so it's going to be a new function, macro, or code duplication between eval/exec.

What do you prefer?

Reply:
Or just make the scope before compiling!

Reply:
Inlined into make_scope which is now called first.